src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLBytes.cs

```diff
@@ -290,6 +290,34 @@ public void SetLength(long value)
             AssertValid();
         }
 
+        private long ReadInternal(Span<byte> buffer, long offset)
+        {
+            if (_state == SqlBytesCharsState.Stream)
+            {
+                if (_stream!.Position != offset)
+                    _stream.Seek(offset, SeekOrigin.Begin);
+                return _stream.Read(buffer);
+            }
+
+            // Adjust count based on data length
+            int count = Math.Min(buffer.Length, (int)(Length - offset));
+
+            Span<byte> span = _rgbBuf!.AsSpan((int)offset, count);
```

Comment:
Instead of the null-forgiving operator I'd prefer a `Debug.Assert(_rgbBuf is not null)`, as in case it's actually `null`, then in CI the assert fails.
Same on other places.

Except it's really 100 % sure that it can't be `null` here, then ignore this comment.

Reply:
I copied all of `!` from old array-based methods so I assume we are 100% sure

Reply:
But internal method also has `IsNull` check under your comment about arguments order 🤔